march_hardware/src/IMotionCube.cpp

```diff
@@ -208,191 +208,196 @@ void IMotionCube::setControlWord(uint16 controlWord)
-                            "triggered. Reset by a Reset Fault "
-                            "command.");
-  bitDescriptions.push_back("\tOver-voltage. Set when protection is triggered. "
-                            "Reset by a Reset Fault command");
-  bitDescriptions.push_back("\tUnder-voltage. Set when protection is "
-                            "triggered. Reset by a Reset Fault command");
-  bitDescriptions.push_back("\tCommand error. This bit is set in several situations. They can be "
-                            "distinguished either by the associated "
-                            "emergency code, or in conjunction with other bits:\n"
-                            "\t\t0xFF03 - Specified homing method not available\n"
-                            "\t\t0xFF04 - A wrong mode is set in object 6060h, modes_of_operation\n"
-                            "\t\t0xFF05 - Specified digital I/O line not available\n"
-                            "\tA function is called during the execution of another function (+ set "
-                            "bit 7 of object 6041h, statusword).\n"
-                            "\tUpdate of operation mode received during a transition. This bit acts "
-                            "just as a warning.");
-  bitDescriptions.push_back("\tDrive disabled due to enable input. Set when "
-                            "enable input is on disable state. Reset "
-                            "when enable input is on enable state");
+    uint16 fiveBitMask = 0b0000000001001111;
+    uint16 sixBitMask = 0b0000000001101111;
+
+    uint16 notReadyToSwitchOn = 0b0000000000000000;
+    uint16 switchOnDisabled = 0b0000000001000000;
+    uint16 readyToSwitchOn = 0b0000000000100001;
+    uint16 switchedOn = 0b0000000000100011;
+    uint16 operationEnabled = 0b0000000000100111;
+    uint16 quickStopActive = 0b0000000000000111;
+    uint16 faultReactionActive = 0b0000000000001111;
+    uint16 fault = 0b0000000000001000;
+
+    uint16 statusWordFiveBitMasked = (statusWord & fiveBitMask);
+    uint16 statusWordSixBitMasked = (statusWord & sixBitMask);
+
+    if (statusWordFiveBitMasked == notReadyToSwitchOn)
+    {
+        return "Not Ready To Switch On";
+    }
+    else if (statusWordFiveBitMasked == switchOnDisabled)
+    {
+        return "Switch On Disabled";
+    }
+    else if (statusWordSixBitMasked == readyToSwitchOn)
+    {
+        return "Ready to Switch On";
+    }
+    else if (statusWordSixBitMasked == switchedOn)
+    {
+        return "Switched On";
+    }
+    else if (statusWordSixBitMasked == operationEnabled)
+    {
+        return "Operation Enabled";
+    }
+    else if (statusWordSixBitMasked == quickStopActive)
+    {
+        return "Quick Stop Active";
+    }
+    else if (statusWordFiveBitMasked == faultReactionActive)
+    {
+        return "Fault Reaction Active";
+    }
+    else if (statusWordFiveBitMasked == fault)
+    {
+        return "Fault";
+    }
+    else return "Not in a recognized IMC state";
```

Comment:
Throw a ROS_WARN here?

Reply:
I've seen this happen before going to operation enabled, so will probably always be thrown before startup